### PR TITLE
chore(argo-cd): refresh from upstream

### PR DESCRIPTION
**Pull request by DrFaust92**
Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-cd/templates/argocd-server/deployment.yaml [hunk 1]
```diff
@@ -299,6 +299,12 @@ spec:
                 name: argocd-cmd-params-cm
                 key: server.enable.proxy.extension
                 optional: true
+          - name: ARGOCD_API_CONTENT_TYPES
```
Comment (DrFaust92) [inline charts/argo-cd/templates/argocd-server/deployment.yaml:302]: added in v2.9.6 change

File: charts/argo-cd/templates/crds/crd-project.yaml [hunk 2]
```diff
@@ -99,16 +99,18 @@ spec:
                   properties:
                     name:
                       description: Name is an alternate way of specifying the target
-                        cluster by its symbolic name
+                        cluster by its symbolic name. This must be set if Server is
```
Comment (DrFaust92) [inline charts/argo-cd/templates/crds/crd-project.yaml:102]: cosmetic 🤷

File: charts/argo-cd/templates/argocd-notifications/clusterrole.yaml [hunk 3]
```diff
@@ -10,13 +10,42 @@ rules:
...
-    - "applications"
+    - applications
+    - appprojects
     verbs:
     - get
     - list
     - watch
     - update
     - patch
+  - apiGroups:
```
Comment (yu-croco) [inline charts/argo-cd/templates/argocd-notifications/clusterrole.yaml:23]: ✏️ 
https://github.com/argoproj/argo-cd/blob/master/manifests/base/notification/argocd-notifications-controller-role.yaml#L29-L44

File: charts/argo-cd/templates/argocd-server/deployment.yaml (hunk 1 above)
Comment (yu-croco) [reply charts/argo-cd/templates/argocd-server/deployment.yaml:302]: ✏️ 
https://github.com/argoproj/argo-cd/blob/v2.9.6/manifests/base/server/argocd-server-deployment.yaml#L230-L235

**Review by yu-croco — APPROVED** [review]
Thank you for your contribution. 🚀

**Review by mbevc1 — APPROVED** [review]

**Review by tico24 — APPROVED** [review]

**Pull request merged**